### PR TITLE
feat: add implementation of divide

### PR DESCRIPTION
**Pull request by Julian-Zankl**
Merge the new changes into the main branch 💯

### CODE REVIEW

**Comment by muhlba91** [comment]
i'd rename the title and the commit message to better explain what happened. for example, `feat: add implementation of divide`.

File: src/test/java/io/muehlbachler/bswe3b/CalculatorServiceTest.java [hunk 1]
```diff
@@ -11,5 +11,18 @@ public void testDividePositiveNumberNoRemainder() {
...
+    }
+
+    @Test
+    public void testDivideNegativeNumberNoRemainder() {
+        assertEquals(1, calc.divide(-1, -1), "-1/-1 should be 1");
+    }
+
+    @Test
+    public void testDivideNegativeNumberWithRemainder() {
+        assertEquals(3, calc.divide(-7, -2), "-7/-2 should be 3");
```
Comment (muhlba91) [inline src/test/java/io/muehlbachler/bswe3b/CalculatorServiceTest.java]: this is not 3 if you look at the specification and the code you wrote.

Comment (Julian-Zankl) [reply src/test/java/io/muehlbachler/bswe3b/CalculatorServiceTest.java]: Has been resolved in https://github.com/muhlba91/fh-burgenland-bswe-3b/pull/5/commits/2f3ffcbf7cbc405a1e82e1d83b0a7bc6e1b16903

File: src/main/java/io/muehlbachler/bswe3b/CalculatorService.java [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
-     * @param a dividend
-     * @param b divisor
-     * @return quotient
+     * Divides a by b.
+     * @param a The dividend
+     * @param b The divisor
+     * @return The quotient
      */
     public int divide(final int a, final int b) {
         return b < 0 ? 0 : a/b;
```
[the remark this answers is not in the document]
Comment (Julian-Zankl) [reply src/main/java/io/muehlbachler/bswe3b/CalculatorService.java]: Proper tests have been added in https://github.com/muhlba91/fh-burgenland-bswe-3b/pull/5/commits/2f3ffcbf7cbc405a1e82e1d83b0a7bc6e1b16903

**Pull request closed without merging**